app/models/owner.js

```diff
@@ -109,6 +109,29 @@ export default VcsEntity.extend({
     return filteredAnnualPlans.sortBy('builds');
   }),
 
+  fetchV2Plans: task(function* () {
+    const url = this.isOrganization ? `/v2_plans_for/organization/${this.id}` : '/v2_plans_for/user';
+    const result = yield this.api.get(url);
+    if (result) {
+      return result.v2_plans.map(plan => {
+        const { store } = this;
+        const planConfig = store.peekRecord('v2-plan-config', plan.id) || store.createRecord('v2-plan-config', {
+          id: plan.id,
+          name: plan.name,
+          startingPrice: plan.starting_price,
+          startingUsers: plan.starting_users,
+          privateCredits: plan.private_credits,
+          publicCredits: plan.public_credits,
+        });
+        return planConfig;
+      });
+    }
```

Comment:
Would it make sense to delegate fetching and extraction to Ember Data? It should be easy to achieve. Just overwrite the `query` hook in `v2-plan-config` adapter, make it accept the type and id parameters and build the request URL based on them. Something like this:
```js
export default V3Adapter.extend({
  buildURL: function (modelName, id, snapshot, requestType, query) {
    const { type, id } = query;
    const isOrganization = type === 'organization';
    const isUser = type === 'user';

    assert("The plan config adapter only supports a query request type", requestType === 'query')
    assert("Invalid request parameters for plan config adapter", isUser || isOrganization && !!id);

    const path = isOrganization ? `v2_plans_for/organization/${id}` : 'v2_plans_for/user';
    return `${this.urlPrefix()}/${path}`
  },

  // This overrides the parent implementation to ignore the query parameters
  query(store, type, query) {
    let url = this.buildURL(type.modelName, null, null, 'query', query);
    return this.ajax(url, 'GET');
  }
});
```

Then Ember Data should extract the data properly using the built-in serializer and store it to the store.

After that, fetching the plan configs would look like this:

```js
const { id, type } = this; // owner properties
const plans = yield this.store.query('v2-plan-config', { type, id })
```

How does it sound to you?

Reply:
I see, this is indeed much cleaner. Thank you for the comment!